mongodb/src/main/java/org/geotools/data/mongodb/MongoDBObjectFeature.java

```diff
@@ -83,8 +83,8 @@ public BoundingBox getBounds() {
 
     @Override
     public Object getDefaultGeometry() {
-        Object o = getDBOValue(mapper.getGeometryPath());
-        return o instanceof DBObject ? mapper.getGeometry((DBObject)o) : null;
+        Object o = mapper.getGeometry(featureDBO);
+        return o instanceof Geometry ? o : null;
```

Comment:
@tbarsballe thanks for the details.  this is strange as this issue appears to be a blocker but the plugin was functional when it went through acceptance testing...